scripts/opposed-wfrp4e.js

```diff
@@ -270,23 +278,35 @@ class OpposedWFRP
   }
 
   // Opposed starting message - manual opposed
-  static createOpposedStartMessage(speaker)
+  static createOpposedStartMessage(speaker, rollMode)
   {
-    ChatMessage.create(
-    {
+    console.log(`Create - Roll Mode selected : ${rollMode}`);
+
+    let chatOptions = {
       user: game.user._id,
       hideData: true,
+      rollMode: rollMode,
       flags:{"opposedStartMessage": true},
       content: `<div><b>${speaker.alias}<b> ${game.i18n.localize("ROLL.OpposedStart")}<div>`
-    }).then(msg => this.startMessage = msg)
+    }
+
+    if (["gmroll", "blindroll"].includes(chatOptions.rollMode)) chatOptions["whisper"] = ChatMessage.getWhisperIDs("GM");
+    if (chatOptions.rollMode === "blindroll") chatOptions["blind"] = true;
+    else if (chatOptions.rollMode === "selfroll") chatOptions["whisper"] = [game.user];
```

Comment:
Can you use `chatDataSetup` to avoid duplicating this code?

Reply:
Ok.